src/main/java/core/basesyntax/Machine.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public abstract class Machine {
+    private String name;
+
+    public abstract void doWork();
+
+    public abstract void stopWork();
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getMachineInfo() {
+        return "name: " + name;
+    }
```

Comment:
The `getMachineInfo` method currently only returns the name of the machine. Consider including the type of the machine (Truck, Bulldozer, Excavator) to make the message more informative as per the checklist.